pymess/utils/logged_requests.py

```diff
@@ -1,9 +1,26 @@
-def generate_session(slug=None, related_objects=None):
+class DefaultTimeoutSessionMixin:
+
+    def __init__(self, timeout=None, **kwargs):
+        super().__init__(**kwargs)
+        self.timeout = timeout
+
+    def request(self, method, url, **kwargs):
+        timeout = kwargs.pop('timeout', self.timeout)
+        return super().request(method, url, timeout=timeout, **kwargs)
+
+
+def generate_session(slug=None, related_objects=None, timeout=None):
     try:
         from security.transport.security_requests import SecuritySession
 
-        return SecuritySession(slug=slug, related_objects=related_objects)
+        class DefaultTimeoutSecuritySession(DefaultTimeoutSessionMixin, SecuritySession):
```

Comment:
why are these defined inside the function?

Reply:
because the funcition generates classes according to installed libraries